scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.codehaus.groovy.control.CompilationFailedException;
+import org.codehaus.groovy.control.CompilerConfiguration;
+import org.objectweb.proactive.Body;
+import org.objectweb.proactive.EndActive;
+import org.objectweb.proactive.InitActive;
+import org.objectweb.proactive.RunActive;
+import org.objectweb.proactive.Service;
+import org.objectweb.proactive.core.body.request.Request;
+import org.objectweb.proactive.core.mop.MethodCallExecutionFailedException;
+import org.objectweb.proactive.extensions.annotation.ActiveObject;
+import org.objectweb.proactive.utils.StackTraceUtil;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.util.TaskLogger;
+
+import groovy.lang.GroovyShell;
+import jdbm.PrimaryHashMap;
+import jdbm.RecordManager;
+import jdbm.RecordManagerFactory;
+
+
+/**
+ * Implementation of the Synchronization service
+ *
+ * This implementation uses an Active Object to allow remote interaction and enforce atomicity
+ *
+ * All requests are <b>synchronous</b> (this is done by returning primitive objects or throwing checked exception)
+ *
+ * Internally, HashMap are used to implement Channels. Channels are persisted thanks to <a href="https://code.google.com/archive/p/jdbm2/">JDBM2</a>
+ *
+ * A custom {@link RunActive#runActivity(Body) runActivity} is implemented to handle wait Methods.
+ * @author ActiveEon Team
+ * @since 26/03/2018
+ */
+@ActiveObject
+public class AOSynchronization implements RunActive, InitActive, EndActive, SynchronizationInternal {
+
+    private static final Logger logger = Logger.getLogger(AOSynchronization.class);
+
+    private static final String RESULT_IS = ", result is: ";
+
+    private static final String QUOTE = "'";
+
+    private static final String PREVIOUS_VALUE_WAS = ", previous value was ";
+
+    private static final String EXCEPTION_WHEN_EVALUATING_CLOSURE = "Exception when evaluating closure : ";
+
+    private static final String ON_KEY = " on key ";
+
+    private static final String WITH_VALUE = ", with value ";
+
+    private File statusFile;
+
+    private File statusFileDirectory;
+
+    private static final String DEFAULT_STORE_SESSION_NAME = "SchedulerStore";
+
+    public static final String STATUS_RECORD_NAME = "STORE";
+
+    private HashMap<String, Channel> inMemoryChannels;
+
+    private PrimaryHashMap<String, Channel> persistedChannels;
+
+    private Queue<Request> waitUntilQueue = new ArrayDeque<>();
```

Comment:
Maybe add a comment to say what this queue is about.

Reply:
ok I will add comments in general for fields in this class